allennlp/data/vocabulary.py

```diff
@@ -76,6 +76,10 @@ def __missing__(self, key: str):
         dict.__setitem__(self, key, value)
         return value
 
+    def add_non_padded_namespaces(self, non_padded_namespaces: Sequence[str]):
+        # add non_padded_namespaces which weren't already present
+        self._non_padded_namespaces.extend(non_padded_namespaces)
+        self._non_padded_namespaces = list(set(self._non_padded_namespaces))
```

Comment:
`self._non_padded_namespaces` was a tuple before, because `DEFAULT_NON_PADDED_NAMESPACES ` are set as tuple. However, when non_padded_namespaces are passed they are lists: `Sequence[str]`.

But its better to make it a set as default and throughout and then `set(list(..))` won't be required. I will change `Sequence[str]` to set and see if everything works. And as you said, following would be updated function:
```
self._non_padded_namespaces.update(non_padded_namespaces)
```
Sounds good?

Reply:
👍